### ISSUE TITLE
Bug: Pattern init fail in release V3.3.1

### ISSUE THREAD

**Issue by LankWang17** [Genivia/RE-flex#176]
Such a pattern string will cause Reflex to hung.
	Pattern pattern(
		"Head\\b|"
		"\\[System\\.Net\\.WebRequest\\]::"
		, "i");

If  changed "Head\\b|" to "Head|", Reflex works well.
it seems strange.

**Comment by LankWang17** [comment]
Sorry, the pattern string is:

**Comment by LankWang17** [comment]
Sorry, the bug string is:
	Pattern pattern(
		"Head\\\\b|"
		"\\\\[System\\\\.Net\\\\.WebRequest\\\\]::"
		, "i");

The html editor makes "\\\\" to "\\".

**Comment by genivia-inc** [comment]
Tried the pattern on MacOS with reflex 3.3.1 and I didn't see anything unusual. Not hanging in the pattern construction. Did you use the pattern to match anything? What OS are you using and what compiler?

**Comment by LankWang17** [comment]
> 

My env: Windows 10, VS2017, Win32-Debug mode. No defined HAVE_AVX2 and HAVE_AVX512BW.

**Comment by LankWang17** [comment]
I used reflex project to test. It hung also. Just like this.
![1678924644350](https://user-images.githubusercontent.com/127750489/225470811-ba9ce78f-0a42-434c-9546-720f381ee3ab.png)

**Comment by genivia-inc** [comment]
This helps to I can test with MSVC++ in x86 mode tomorrow. Strange if there is somehow a difference with other OS. Or perhaps it's a 32 bit build problem.

**Comment by LankWang17** [comment]
![image](https://user-images.githubusercontent.com/127750489/225474954-5d65eb32-7e91-4d01-a2a5-35229b4f29e4.png)

**Comment by LankWang17** [comment]
OK. Thank you.

**Comment by genivia-inc** [comment]
I see the same thing. It is the combination of pattern option `"i"` with a word boundary marker `\b`. This exponentially grows the DFA to 2,622,046 nodes and 3,146,356 edges (!) I will further investigate the cause of this. This may take a bit of time to work on. A POSIX DFA can exponentially grow in the worst case for certain regex patterns that are unlikely to be used in practice and that should not happen in this case.

**Comment by genivia-inc** [comment]
This works OK as a temporary workaround, placing a `\b` after the second pattern:
~~~
Pattern pattern(
"Head\\b|"
"\\[System\\.Net\\.WebRequest\\]::\\b"
, "i");
~~~

**Comment by LankWang17** [comment]
> This works OK as a temporary workaround, placing a `\b` after the second pattern:
> 
> ```
> Pattern pattern(
> "Head\\b|"
> "\\[System\\.Net\\.WebRequest\\]::\\b"
> , "i");
> ```

OK, I'll try it.

**Comment by genivia-inc** [comment]
I found a subtle bug in the DFA constructor. There are a lot of unused DFA fragments generated for this specific case that triggers this bug. There is nothing intrinsically wrong with the pattern that should not explode the DFA in size. Will fix it and release an update after testing.

**Comment by genivia-inc** [comment]
Fixed in release 3.3.2.

**Comment by genivia-inc** [comment]
Thanks for reporting this performance issue. It is very useful to receive feedback. We test the software a lot. Nevertheless, it can happen that something does not work perfectly.
